### PR TITLE
User Peers and User related TI

### PR DESCRIPTION
**Pull request by YaronFruchtmann**
Adds two new insights

## Proposed Changes

  - User peers to list the user closest peers
  - Uer related TI
  -

### CODE REVIEW

File: Insights/Account/UserAccessPermissions.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    let userSPAccess = userAccess| where TargetEntityType == 'ServicePrincipal' | project SourceEntityName, UserId = SourceEntityId, ServicePrincipalId = TargetEntityId, TimeGenerated | summarize arg_max(TimeGenerated, *) by ServicePrincipalId;
+    let userSPSubAccess = userSPAccess | join kind = inner UserAccessAnalytics on $left.ServicePrincipalId == $right.SourceEntityId | where TargetEntityType == 'AzureSubscription' and TimeGenerated <= blastRadTime | project SourceEntityName, UserId, GroupId = '', ServicePrincipalId, TimeGenerated, SubscriptionName = TargetEntityName, Subscription = TargetEntityId, Role = AccessLevel, AccessType| summarize arg_max(TimeGenerated, *) by ServicePrincipalId, Subscription, Role; 
+    userGroupSubAccess | union kind=outer userSubAccess | union kind=outer userSPSubAccess | extend Role = iff(Role == '_UNKNOWN_RBAC_ROLE_', 'Other', Role) };
```
Comment (shainw) [inline Insights/Account/UserAccessPermissions.yaml:16]: Two items:

1. The union doesn't need to be two statements, can be union kind=outer userSubAccess, userSPSubAccess.
2. Why write in Other for the unknown rbac role, I think it is more important to match the log in this case as we are not group a bunch of items to Other, just simply renaming?

File: Insights/Account/UserAccessPermissions.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
+SchemaVersion: '1.0'
+DataTypes:
+  - DataType: UserAccessAnalytics
+Provider: Sentinel
+Type: KQL
+EntitiesFilter: {}
+BaseQuery: |
+  let GetUserSubscriptionRbac = (v_Account_AadUserId:string) { 
+    let blastRadTime = now();
+    let userAccess = UserAccessAnalytics|where SourceEntityId == v_Account_AadUserId;
```
Comment (shainw) [inline Insights/Account/UserAccessPermissions.yaml:10]: In general, these are difficult to read, would be good to have a line return for each new statement.

File: Insights/Account/AccountPeers.yaml [hunk 3]
```diff
@@ -0,0 +1,41 @@
+SchemaVersion: '1.0'
+DataTypes:
+- DataType: UserPeerAnalytics
+Provider: Sentinel
+Type: KQL
+EntitiesFilter: {}
+BaseQuery: |
+  let GetUserPeers = (v_Account_Name:string, v_Account_UPNSuffix:string) { UserPeerAnalytics | extend Account_UPN = strcat(v_Account_Name, '@',v_Account_UPNSuffix) | where UserPrincipalName == Account_UPN};
```
Comment (shainw) [inline Insights/Account/AccountPeers.yaml:8]: New line return for each new statement would help readability.

File: Insights/Account/UserTI.yaml [hunk 4]
```diff
@@ -0,0 +1,47 @@
+SchemaVersion: '1.0'
+DataTypes:
+- DataType: BehaviorAnalytics
+Provider: Sentinel
+Type: KQL
+EntitiesFilter: {}
+BaseQuery: |
+  let UserTI=(v_Account_Name:string, v_Account_UPNSuffix){
```
Comment (shainw) [inline Insights/Account/UserTI.yaml:8]: missing :string on v_Account_UPNSuffix

File: Insights/URL/URL_TI.yaml [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+BaseQuery: | 
+  let UrlThreatIntel=(URL_Url:string){
+  ThreatIntelligenceIndicator 
+  | where isnotempty(Url) or isnotempty( DomainName )
+  | extend match = Url matches regex @'^\\w{2,6}://' 
+  | extend Url = iff(match, Url, strcat('Unknown://', Url))
+  | extend parsedURL=parse_url(Url) 
+  | evaluate bag_unpack(parsedURL) 
+  | where parse_url(URL_Url)['Host'] == coalesce(tostring(parse_url(Url)['Host']),DomainName, Url) 
+  | parse Description with * 'iType: ' MaliciousType'; State:' State'; Org: 'Org'; Source: 'Source_"|;
```
Comment (shainw) [inline Insights/URL/URL_TI.yaml:16]: This line is broken, not sure exactly what you were trying to accomplish so I cannot comment on a suitable fix.

File: Insights/URL/URL_TI.yaml [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  }
```
Comment (shainw) [inline Insights/URL/URL_TI.yaml:17]: need a semi-colon here.

**Review by shainw — CHANGES_REQUESTED** [review]

File: Insights/Account/UserAccessPermissions.yaml (hunk 2 above)
Comment (YaronFruchtmann) [reply Insights/Account/UserAccessPermissions.yaml:10]: agree. added newlines

File: Insights/Account/UserAccessPermissions.yaml (hunk 1 above)
Comment (YaronFruchtmann) [reply Insights/Account/UserAccessPermissions.yaml]: do not understand (2)

**Review by shainw — APPROVED** [review]
One question on output, but not a blocker.  Approved.

**Pull request merged**